### PR TITLE
docs - resource groups catalog table additions/updates

### PR DESCRIPTION
**Pull request by lisakowen**
create 2 new, update 3 existing, system catalog tables for resource group support:
- pg_authid - add rolresgroup column
- pg_resgroup - new catalog table
- pg_resgroupcapability - new catalog table
- pg_roles - add rolresgroup column
- pg_stat_activity - add rsgid, rsgname, rsgqueueduration

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1" xml:lang="en"><title id="hi141670">pg_resgroup</title><body><p>The <codeph>pg_resgroup</codeph> system catalog table contains information about Greenplum Database resource groups, which are used for managing concurrent statements, CPU, and memory resources.
+      This table is populated only on the Greenplum Database master. This table, defined in the
```
Comment (xiong-gang) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml]: Currently it's populated on both master and segments. We will evaluate the possibility to remove it from segment later.

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroupcapability.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1" xml:lang="en">
+  <title id="hk141670">pg_resgroupcapability</title>
+  <body>
+    <p>The <codeph>pg_resgroupcapability</codeph> system catalog table contains information about
+      the capabilities and limits of defined Greenplum Database resource groups.
+      You can join this table to the <xref
+        href="pg_resgroup.xml#topic1" type="topic" format="dita"/> table by resource group object
+      ID.</p>
+    <p>The <codeph>pg_resgroupcapability</codeph> table is populated only on the Greenplum Database
```
Comment (xiong-gang) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroupcapability.xml]: Same as pg_resgroup, it's on both master and segments

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_stat_activity.xml [hunk 3]
```diff
@@ -9,4 +9,8 @@ these columns are only visible if the user examining the view is a superuser
...
-can be:<p>lock replication</p></entry></row></tbody></tgroup></table></body></topic>
\ No newline at end of file
+can be:<p>lock replication</p></entry></row>
```
Comment (xiong-gang) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_stat_activity.xml]: we added a new 'reason' `g` represent resource group

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroupcapability.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+              referenced. </entry>
+          </row>
+          <row>
+            <entry colname="col1">
+              <codeph>proposed</codeph>
+            </entry>
+            <entry colname="col2">opaque type</entry>
+            <entry colname="col3"/>
+            <entry colname="col4">If the limit cannot be updated immediately, the proposed value
+              for the resource limit referenced in this record.</entry>
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroupcapability.xml]: What is the value otherwise?

**Comment by mkiyama** [comment]
In the pg_resgroup, pg_resqueue topics do you want to mention the GUC that controls
whether groups or queues are "active"?

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml (hunk 1 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml]: thanks for reviewing, @xiong-gang.  for future reference how would i tell that a table is populated on both master and segments?  i.e. is there a query i could perform?  or some config/etc i could examine?

**Comment by lisakowen** [comment]
@mkiyama - re: adding info about the gp_resource_manager GUC...  i agree this GUC should be mentioned in these topics.  we are adding the resource group docs in stages, GUC docs coming soon.  i think it makes sense to wait til we actually document this GUC and how to enable resource groups before mentioning it anywhere.

**Comment by lisakowen** [comment]
updates per review comments:
- pg_stat_activity - add resgroup to list of waiting_reason s
- pg_resgroup* - remove references to where the tables reside
- pg_resgroupcapability - clarified that the proposed field reflects the current value when no alter is pending

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1" xml:lang="en"><title id="hi141670">pg_resgroup</title><body><p>The <codeph>pg_resgroup</codeph> system catalog table contains information about Greenplum Database resource groups, which are used for managing concurrent statements, CPU, and memory resources.
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml]: Should we be including "memory" at this point?

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+      This table, defined in the
+        <codeph>pg_global</codeph> tablespace, is globally shared across all databases in
+      the system.</p><table id="hi141982"><title>pg_catalog.pg_resgroup</title><tgroup cols="4"><colspec colnum="1" colname="col1" colwidth="131pt"/><colspec colnum="2" colname="col2" colwidth="86pt"/><colspec colnum="3" colname="col3" colwidth="85pt"/><colspec colnum="4" colname="col4" colwidth="147pt"/><thead><row><entry colname="col1">column</entry><entry colname="col2">type</entry><entry colname="col3">references</entry><entry colname="col4">description</entry></row></thead><tbody><row><entry colname="col1"><codeph>rsgname</codeph></entry><entry colname="col2">name</entry><entry colname="col3"/><entry colname="col4">The name of the resource group.</entry></row><row><entry colname="col1"><codeph>parent</codeph></entry><entry colname="col2">oid</entry><entry colname="col3"/><entry colname="col4">Unused; reserved for future use.</entry></row>
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml]: Might make sense to load this up into oxygen to improve the formatting a bit..

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml (hunk 5 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml]: yes, memory-related resource mgmt is currently under development.  when you create a resource group, you must provide the memory_limit, so some aspects of memory mgmt are working.

File: gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml (hunk 6 above)
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/system_catalogs/pg_resgroup.xml]: sure.  i used the pg_resqueue page as a template for the pg_resgroup page, maybe i should load that file up as well ...

**Pull request merged**